packages/twenty-front/src/modules/workflow/hooks/useFindWorkflowWithCurrentVersion.tsx

```diff
@@ -0,0 +1,71 @@
+import { CoreObjectNameSingular } from '@/object-metadata/types/CoreObjectNameSingular';
+import { useFindManyRecords } from '@/object-record/hooks/useFindManyRecords';
+import { useFindOneRecord } from '@/object-record/hooks/useFindOneRecord';
+import {
+  Workflow,
+  WorkflowVersion,
+  WorkflowWithCurrentVersion,
+} from '@/workflow/types/Workflow';
+import { isDefined } from 'twenty-ui';
+
+export const useFindWorkflowWithCurrentVersion = (
+  workflowId: string | undefined,
+): WorkflowWithCurrentVersion | undefined => {
+  const { record: workflow } = useFindOneRecord<Workflow>({
+    objectNameSingular: CoreObjectNameSingular.Workflow,
+    objectRecordId: workflowId,
+    recordGqlFields: {
+      id: true,
+      name: true,
+      lastPublishedVersionId: true,
+      statuses: true,
+      versions: true,
+    },
+    skip: !isDefined(workflowId),
+  });
+
+  const { record: lastPublishedVersion } = useFindOneRecord<WorkflowVersion>({
+    objectNameSingular: CoreObjectNameSingular.WorkflowVersion,
+    objectRecordId: workflow?.lastPublishedVersionId,
+    skip: !(
+      typeof workflow?.lastPublishedVersionId === 'string' &&
+      workflow.lastPublishedVersionId !== ''
+    ),
+    recordGqlFields: {
+      id: true,
+      name: true,
+      createdAt: true,
+      updatedAt: true,
+      workflowId: true,
+      trigger: true,
+      steps: true,
+    },
+  });
+
+  const { records: workflowDraftVersions } =
+    useFindManyRecords<WorkflowVersion>({
+      objectNameSingular: CoreObjectNameSingular.WorkflowVersion,
+      filter: {
+        status: {
+          eq: 'DRAFT',
+        },
+        workflowId: {
+          eq: workflowId,
+        },
+      },
+      limit: 1,
+      skip: !isDefined(workflowId),
+    });
+
+  if (!isDefined(workflow)) {
```

Comment:
move that check just after `useFindOneRecord<Workflow>(...)` to avoid useless calls

Reply:
I think that wouldn't work because it would break one of the rules of React hooks: _hooks mustn't be called conditionally_.

The `skip` parameter should prevent unnecessary work.